Sources/GRPCHTTP2Core/Client/GRPCClientStreamHandler.swift

```diff
@@ -143,23 +143,16 @@ extension GRPCClientStreamHandler {
       do {
         self.flushPending = true
         let headers = try self.stateMachine.send(metadata: metadata)
-        context.write(self.wrapOutboundOut(.headers(.init(headers: headers))), promise: nil)
-        // TODO: move the promise handling into the state machine
-        promise?.succeed()
+        context.write(self.wrapOutboundOut(.headers(.init(headers: headers))), promise: promise)
       } catch {
         context.fireErrorCaught(error)
-        // TODO: move the promise handling into the state machine
-        promise?.fail(error)
```

Comment:
I think we _do_ need this one: we only reach here if we're in an invalid state and the state machine doesn't take care of the promise in that situation.

Reply:
Yes you're right, I fixed this in a couple of other places before but I missed these two. Good catch!